### PR TITLE
pkgs/tools/compression: cleanup of various packages

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Motivation for this change
A ton of these packages are fairly outdated, with a lot of formatting issues. This fixes some of those.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by RaghavSood** [comment]
Result of `nixpkgs-review pr 104849` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>23 packages built:</summary>
  <ul>
    <li>ark (kdeApplications.ark ,libsForQt5.ark)</li>
    <li>automysqlbackup</li>
    <li>bsc</li>
    <li>bsdiff</li>
    <li>dtrx</li>
    <li>globulation2</li>
    <li>gzrt</li>
    <li>hydra-unstable</li>
    <li>kzipmix</li>
    <li>lbzip2</li>
    <li>lhasa</li>
    <li>libsForQt512.ark</li>
    <li>libsForQt514.ark</li>
    <li>lrzip</li>
    <li>lzop</li>
    <li>milkytracker</li>
    <li>pigz</li>
    <li>pixz</li>
    <li>pxz</li>
    <li>rzip</li>
    <li>sanoid</li>
    <li>wal_e</li>
    <li>zsync</li>
  </ul>
</details>

File: pkgs/tools/compression/bsdiff/default.nix [hunk 1]
```diff
@@ -5,12 +5,11 @@ stdenv.mkDerivation rec {
   version = "4.3";
 
   src = fetchurl {
-    url    = "https://www.daemonology.net/bsdiff/${pname}-${version}.tar.gz";
+    url = "https://www.daemonology.net/bsdiff/${pname}-${version}.tar.gz";
     sha256 = "0j2zm3z271x5aw63mwhr3vymzn45p2vvrlrpm9cz2nywna41b0hq";
   };
 
   buildInputs = [ bzip2 ];
-  patches = [ ./include-systypes.patch ];
```
Comment (ajs124) [inline pkgs/tools/compression/bsdiff/default.nix:13]: When and why was this patch added and why is it not needed anymore?

File: pkgs/tools/compression/pxz/default.nix [hunk 2]
```diff
@@ -1,44 +1,34 @@
...
+  src = fetchFromGitHub {
+    owner = "jnovy";
+    repo = "pxz";
+    rev = "124382a6d0832b13b7c091f72264f8f3f463070a";
+    sha256 = "15mmv832iqsqwigidvwnf0nyivxf0y8m22j2szy4h0xr76x4z21m";
   };
 
   buildInputs = [ lzma ];
 
-  patches = [ ./_SC_ARG_MAX.patch ];
```
Comment (ajs124) [inline pkgs/tools/compression/pxz/default.nix:17]: Same question for this patch.

File: pkgs/tools/compression/pigz/default.nix [hunk 3]
```diff
@@ -1,31 +1,27 @@
-{ stdenv, fetchurl, zlib, util-linux }:
+{ stdenv, fetchurl, zlib, utillinux }:
```
Comment (r-burns) [inline pkgs/tools/compression/pigz/default.nix:1]: https://github.com/NixOS/nixpkgs/pull/104776

**Comment by r-burns** [comment]
I agree with the pname/version and fetchFromGitHub changes but a lot of the style changes just look like diff churn to me. I agree with your style choices but I don't think you should be forcing them on these packages' maintainers.

File: pkgs/tools/compression/brotli/default.nix [hunk 4]
```diff
@@ -22,8 +22,7 @@ stdenv.mkDerivation rec {
     sha256 = "sOeXNVsCaBSD9i82GRUDrkyreGeQ7qaJWjjy/uLL0/0=";
   });
 
-  cmakeFlags = []
-    ++ stdenv.lib.optional staticOnly "-DBUILD_SHARED_LIBS=OFF";
+  cmakeFlags = stdenv.lib.optional staticOnly "-DBUILD_SHARED_LIBS=OFF";
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/brotli/default.nix:25]:
```suggestion
  cmakeFlags = lib.optional staticOnly "-DBUILD_SHARED_LIBS=OFF";
```

File: pkgs/tools/compression/brotli/default.nix [hunk 5]
```diff
@@ -52,9 +51,7 @@ stdenv.mkDerivation rec {
 
   meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/brotli/default.nix:52]:
```suggestion
  meta = with lib; {
```

File: pkgs/tools/compression/bsc/default.nix [hunk 6]
```diff
@@ -1,26 +1,25 @@
...
-        --replace 'g++' '$(CXX)'
+    substituteInPlace makefile --replace 'g++' '$(CXX)'
   '';
 
-  preInstall = ''
-    makeFlagsArray+=("PREFIX=$out")
-  '';
+  makeFlags = [ "PREFIX=$(out)" ];
 
   meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/bsc/default.nix:24]:
```suggestion
  meta = with lib; {
```

File: pkgs/tools/compression/bsdiff/default.nix [hunk 7]
```diff
@@ -20,18 +19,15 @@ stdenv.mkDerivation rec {
...
-    cp bsdiff    $out/bin
-    cp bspatch   $out/bin
-    cp bsdiff.1  $out/share/man/man1
-    cp bspatch.1 $out/share/man/man1
+    cp {bsdiff,bspatch} $out/bin
+    cp {bsdiff.1,bspatch.1} $out/share/man/man1
   '';
 
-  meta = {
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/bsdiff/default.nix:26]:
```suggestion
  meta = with lib; {
```

File: pkgs/tools/compression/dejsonlz4/default.nix [hunk 8]
```diff
@@ -1,28 +1,30 @@
...
 
-    meta = with stdenv.lib; {
-      description = "Decompress Mozilla Firefox bookmarks backup files";
-      homepage = "https://github.com/avih/dejsonlz4";
-      license = licenses.bsd2;
-      maintainers = with maintainers; [ mt-caret ];
-      platforms = platforms.linux;
-    };
-  }
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/dejsonlz4/default.nix:23]:
```suggestion
  meta = with lib; {
```

File: pkgs/tools/compression/gzrt/default.nix [hunk 9]
```diff
@@ -15,10 +16,10 @@ stdenv.mkDerivation rec {
     cp gzrecover $out/bin
   '';
 
-  meta = {
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/gzrt/default.nix:19]:
```suggestion
  meta = with lib; {
```

File: pkgs/tools/compression/kzipmix/default.nix [hunk 10]
```diff
@@ -1,25 +1,25 @@
...
     mkdir -p $out/bin
-    cp kzip zipmix $out/bin
-    
-    patchelf --set-interpreter ${stdenv.glibc.out}/lib/ld-linux.so.2 $out/bin/kzip
-    patchelf --set-interpreter ${stdenv.glibc.out}/lib/ld-linux.so.2 $out/bin/zipmix
+    cp amd64/{kzip,zipmix} $out/bin
   '';
 
-  meta = {
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/kzipmix/default.nix:19]:
```suggestion
  meta = with lib; {
```

File: pkgs/tools/compression/lhasa/default.nix [hunk 11]
```diff
@@ -1,21 +1,24 @@
...
-stdenv.mkDerivation {
-  name = "lhasa-0.3.1";
   src = fetchurl {
-    url = "https://soulsphere.org/projects/lhasa/lhasa-0.3.1.tar.gz";
+    url = "https://soulsphere.org/projects/lhasa/lhasa-${version}.tar.gz";
     sha256 = "092zi9av18ma20c6h9448k0bapvx2plnp292741dvfd9hmgqxc1z";
   };
-  meta = {
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/lhasa/default.nix:12]:
```suggestion
  meta = with lib; {
```

File: pkgs/tools/compression/lrzip/default.nix [hunk 12]
```diff
@@ -9,13 +9,13 @@ stdenv.mkDerivation rec {
     sha256 = "0mb449vmmwpkalq732jdyginvql57nxyd31sszb108yps1lf448d";
   };
 
-  buildInputs = [ zlib lzo bzip2 nasm perl ];
+  nativeBuildInputs = [ nasm perl ];
+  buildInputs = [ zlib lzo bzip2 ];
 
-  meta = {
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/lrzip/default.nix:15]:
```suggestion
  meta = with lib; {
```

File: pkgs/tools/compression/pigz/default.nix [hunk 13]
```diff
@@ -1,31 +1,27 @@
...
   src = fetchurl {
-    url = "https://www.zlib.net/${name}/${name}-${version}.tar.gz";
+    url = "https://www.zlib.net/pigz/pigz-${version}.tar.gz";
     sha256 = "0wsgw5vwl23jrnpsvd8v3xcp5k4waw5mk0164fynjhkv58i1dy54";
   };
 
   enableParallelBuilding = true;
 
-  buildInputs = [zlib] ++ stdenv.lib.optional stdenv.isLinux util-linux;
+  buildInputs = [ zlib ] ++ stdenv.lib.optional stdenv.isLinux utillinux;
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/pigz/default.nix:14]:
```suggestion
  buildInputs = [ zlib ] ++ lib.optional stdenv.isLinux utillinux;
```

File: pkgs/tools/compression/pigz/default.nix [hunk 14]
```diff
@@ -1,31 +1,27 @@
...
-      ln -s pigz.1 "$out/share/man/man1/unpigz.1"
-      install -Dm755 pigz.pdf "$out/share/doc/pigz/pigz.pdf"
+
+  installPhase = ''
+    install -Dm755 pigz "$out/bin/pigz"
+    install -Dm755 pigz.1 "$out/share/man/man1/pigz.1"
+    install -Dm755 pigz.pdf "$out/share/doc/pigz/pigz.pdf"
   '';
 
   meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/pigz/default.nix:27]:
```suggestion
  meta = with lib; {
```

File: pkgs/tools/compression/pixz/default.nix [hunk 15]
```diff
@@ -1,37 +1,35 @@
...
     ./autogen.sh
   '';
 
-  meta = {
-    inherit version;
-    description = ''A parallel compressor/decompressor for xz format'';
-    license = stdenv.lib.licenses.bsd2;
-    maintainers = [stdenv.lib.maintainers.raskin];
-    platforms = stdenv.lib.platforms.unix;
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/pixz/default.nix:29]:
```suggestion
  meta = with lib; {
```

File: pkgs/tools/compression/pxz/default.nix [hunk 16]
```diff
@@ -1,44 +1,34 @@
...
-  '';
-
   installPhase = ''
     mkdir -p $out/bin $out/share/man/man1
     cp pxz $out/bin
     cp pxz.1 $out/share/man/man1
   '';
 
-  meta = {
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/pxz/default.nix:22]:
```suggestion
  meta = with lib; {
```

File: pkgs/tools/compression/rzip/default.nix [hunk 17]
```diff
@@ -16,10 +19,10 @@ stdenv.mkDerivation {
     })
   ];
 
-  meta = {
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/rzip/default.nix:22]:
```suggestion
  meta = with lib; {
```

File: pkgs/tools/compression/zsync/default.nix [hunk 18]
```diff
@@ -1,20 +1,21 @@
...
   src = fetchurl {
-    url = "http://zsync.moria.org.uk/download/${name}.tar.bz2";
+    url = "http://zsync.moria.org.uk/download/zsync-${version}.tar.bz2";
     sha256 = "1wjslvfy76szf0mgg2i9y9q30858xyjn6v2acc24zal76d1m778b";
   };
 
   makeFlags = [ "AR=${stdenv.cc.bintools.targetPrefix}ar" ];
 
-  meta = {
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/zsync/default.nix:14]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by ajs124** [comment]
I've rebased this here https://github.com/helsinki-systems/nixpkgs/tree/cleanup/compression, if you want to still have this merged @IvarWithoutBones

**Comment by SuperSandro2000** [comment]
Closed in favor of #130928

**Comment by IvarWithoutBones** [comment]
Thanks @ajs124! Totally forgot about this one.

**Pull request closed without merging**